File: test-suite/src/test/java/io/micronaut/views/model/ConfigViewModelProcessor.java
package io.micronaut.views.model;

import io.micronaut.context.annotation.Requires;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.http.HttpRequest;
import io.micronaut.runtime.ApplicationConfiguration;
import io.micronaut.views.ModelAndView;
import javax.inject.Inject;
import javax.inject.Singleton;

import java.util.Map;

@Requires(property = "spec.name", value = "ModelAndViewSpec")
//tag::class[]
@Singleton // <1>
public class ConfigViewModelProcessor implements ViewModelProcessor {
    private final ApplicationConfiguration config;

    ConfigViewModelProcessor(ApplicationConfiguration environment) {
        this.config = environment;
    }

    @Override
    public void process(@NonNull HttpRequest<?> request,
                        @NonNull ModelAndView<Map<String, Object>> modelAndView) {
        modelAndView.getModel()
                .ifPresent(model -> model.put("config", config));
    }
}
//end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/model/ConfigViewModelProcessor.kt
package io.micronaut.views.model

import io.micronaut.context.annotation.Requires
import io.micronaut.http.HttpRequest
import io.micronaut.runtime.ApplicationConfiguration
import io.micronaut.views.ModelAndView
import javax.inject.Singleton

@Requires(property = "spec.name", value = "ModelAndViewSpec")
//tag::class[]
@Singleton // <1>
class ConfigViewModelProcessor internal constructor(private val config: ApplicationConfiguration) :
    ViewModelProcessor<MutableMap<String, Any>> {

    override fun process(request: HttpRequest<*>, modelAndView: ModelAndView<MutableMap<String, Any>>) {
        modelAndView.model
            .ifPresent { model ->
                config.name.ifPresent { name -> model["applicationName"] = name }
            }
    }
}
//end::class[]
